src/utils.jl

```diff
@@ -163,3 +163,216 @@ cangetdocs(mod::Module, word::Symbol) =
   !Base.isdeprecated(mod, word) &&
   isdefined(mod, Symbol("@doc"))
 cangetdocs(mod::Module, word::String) = cangetdocs(mod, Symbol(word))
+
+#=
+module file detections
+
+adapted from https://github.com/timholy/Revise.jl/tree/b0c5c864ea78b93caaa820cb9cfc45eca47f43ff
+=#
+
+using Base: PkgId, UUID
+
+"""
+    parentfile, included_files = modulefiles(mod::Module)
+
+Return the `parentfile` in which `mod` was defined, as well as a list of any
+other files that were `include`d to define `mod`. If this operation is unsuccessful,
+`(nothing, nothing)` is returned.
+All files are returned as absolute paths.
+"""
+function modulefiles(mod::Module)
+  # NOTE: src_file_key stuff was removed when adapted
+  parentfile = String(first(methods(getfield(mod, :eval))).file)
+  id = Base.PkgId(mod)
+  if id.name == "Base" || Symbol(id.name) ∈ stdlib_names
+    parentfile = normpath(Base.find_source_file(parentfile))
+    filedata = Base._included_files
+  else
+    use_compiled_modules() || return nothing, nothing   # FIXME: support non-precompiled packages
+    _, filedata = pkg_fileinfo(id)
+  end
+  filedata === nothing && return nothing, nothing
+  included_files = filter(mf -> mf[1] == mod, filedata)
+  return fixpath(parentfile), [fixpath(mf[2]) for mf in included_files]
+end
```

Comment:
All the adaption is for `modulefiles` and `parentfile`.

The `src_file_key` stuff is removed, since every modifed files will be caught by `regeneratesymbols` handler.
The other changes are minor, basically making every path to be full path